src/frontend/Player/createVideojsPlayer.ts

```diff
@@ -0,0 +1,168 @@
+import 'videojs-contrib-quality-levels';
+import 'videojs-http-source-selector';
+
+import { appData, getDecodedJwt } from '../data/appData';
+import { liveState, Video, videoSize } from '../types/tracks';
+import {
+  InitializedContextExtensions,
+  InteractedContextExtensions,
+} from '../types/XAPI';
+import { report } from '../utils/errors/report';
+import { Nullable } from '../utils/types';
+import { XAPIStatement } from '../XAPI/XAPIStatement';
+
+import { intl } from '../index';
+
+export const createVideojsPlayer = (
+  videoNode: HTMLVideoElement,
+  dispatchPlayerTimeUpdate: (time: number) => void,
+  video: Video,
+): VideoJsPlayer => {
+  // add the video-js class name to the video attribute.
+  videoNode.classList.add('video-js', 'vjs-big-play-centered');
+  const resolutions = Object.keys(video.urls.mp4).map(
+    (size) => Number(size) as videoSize,
+  );
+
+  const player = videojs(videoNode, {
+    controls: true,
+    debug: false,
+    fluid: true,
+    language: intl.locale,
+    liveui: video.live_state === liveState.RUNNING,
+    playbackRates: [0.5, 0.75, 1, 1.25, 1.5, 1.75, 2, 4],
+    plugins: {
+      httpSourceSelector: {
+        default: 'low',
+      },
+    },
+    responsive: true,
+  });
+  player.httpSourceSelector();
+  const qualityLevels = player.qualityLevels();
+  const tracks = player.remoteTextTracks();
+
+  /************************** XAPI **************************/
+
+  let xapiStatement: XAPIStatement;
+  try {
+    xapiStatement = new XAPIStatement(appData.jwt!, getDecodedJwt().session_id);
+  } catch (error) {
+    report(error);
+    throw error;
+  }
+
+  let currentTime: number = 0;
+  let seekingAt: number = 0;
+  let hasSeeked: boolean = false;
+  let isInitialized: boolean = false;
+
+  const getCurrentTrack = (): Nullable<TextTrack> => {
+    // TextTrackList is not an iterable object
+    // tslint:disable-next-line:prefer-for-of
+    for (let i = 0; i < tracks.length; i++) {
+      if (tracks[i].mode === 'showing') {
+        return tracks[i];
+      }
+    }
+
+    return null;
+  };
+  let currentTrack = getCurrentTrack();
+
+  player.on('canplaythrough', () => {
+    if (isInitialized) {
+      return;
+    }
+
+    const contextExtensions: InitializedContextExtensions = {
+      ccSubtitleEnabled: currentTrack !== null,
+      fullScreen: player.isFullscreen(),
+      length: player.duration(),
+      speed: `${player.playbackRate()}x`,
+      volume: player.volume(),
+    };
+
+    xapiStatement.initialized(contextExtensions);
+    isInitialized = true;
+  });
+
+  player.on('playing', () => {
+    xapiStatement.played({
+      time: player.currentTime(),
+    });
+  });
+
+  player.on('pause', () => {
+    xapiStatement.paused({
+      time: player.currentTime(),
+    });
+  });
+
+  /**************** Seeked statement ***********************/
+
+  player.on('timeupdate', () => {
+    if (true === isInitialized && false === player.seeking()) {
```

Comment:
Really needs to check these values are booleans?
```suggestion
    if (isInitialized && !player.seeking()) {
```

Reply:
No it's an old backend develop habit 😁